Vegetarian-Food-Finder/Model/TimeZoneConverter.swift

```diff
@@ -0,0 +1,180 @@
+import Foundation
+import UIKit
+
+/// A class that is responsible for date manuplation so the user can make events at their desiered time
+final class TimeZoneConverter {
+    let selectedRestaurant: BusinessDetails
+    init(
+        selectedRestaurant: BusinessDetails
+    ) {
+        self.selectedRestaurant = selectedRestaurant
+    }
+
+    /// computed property that represents the TimeZone of the user
+    private var userCurrnetTime: TimeZone {
+        return TimeZone.current
+    }
+
+    /// Enum that handles the case if the restaurant is closed
+    enum PickedDateError: Error {
+        case invalidDate
+    }
+
+    /**
+      gets the restaurants TimeZone by using reverseGeocoding for locations
+
+     - parameter  completion handler: Gives back the TimeZone of the restaurant
+
+     - Returns: TimeZone of the restaurant
+     */
+    func restaurantLocalTime(
+        completion: @escaping
+        (TimeZone, Error) -> Void
+    ) {
+        let location = CLLocation(
+            latitude: selectedRestaurant.coordinates.latitude,
+            longitude: selectedRestaurant.coordinates.longitude)
+        let geoCoder = CLGeocoder()
+        geoCoder.reverseGeocodeLocation(location) { placeMarks, error in
+            if let placeMark = placeMarks?[0].timeZone, let err = error {
+                completion(placeMark, err)
+            }
+        }
+    }
+
+    /**
+      gets the number of days that restaurant is open
+
+     - Returns: an array of int
+     */
+    func getDayNumber(
+    ) -> [Int] {
+        var pickedDay = [0]
+        let restaurantWorkingHoursDetails = selectedRestaurant.hours.weekDetails
+        for dayNumber in restaurantWorkingHoursDetails {
+            pickedDay.append(dayNumber.day)
+        }
+        return pickedDay
+    }
+
+    /**
+      Checks if the restaurant is open at the date the user picked
+
+     - parameter  pickedDate: UIDatePicker representing the date the user picked
+
+     returns dictionary of Bool(represents If restaurant open) and Int(day that it's open)
+     */
+    func checkIfOpenAt(
+        pickedDate: UIDatePicker
+    ) -> [Bool: Int] {
+        let index = Calendar.current.component(.weekday, from: pickedDate.date)
+        let weekDays = getDayNumber()
+        for day in weekDays {
+            if day == index {
+                return [true: day]
+            }
+        }
+        return [false: 0]
+    }
+
+    /**
+      converts the given Time from the Api(String) to Int of hour and minutes
+
+     - parameter  timeString: Time String that is provided by Yelp
+
+     returns dictionary of Int(represents the Hour) and Int(represents the Minutes)
+     */
+    func convertToTimeInt(
+        timeString: String
+    ) -> [Int: Int] {
+        var intgerTime = [0: 0]
+        if let hour = Int(timeString.prefix(2)),
+           let minutes = Int(timeString.suffix(2)) {
+            intgerTime = [hour: minutes]
+        }
+        return intgerTime
```

Comment:
There is  A type Calendar.Component where you can store the hours and minutes (and more if needed)